services/tables/src/main/java/com/linkedin/openhouse/tables/repository/impl/OpenHouseInternalRepositoryImpl.java

```diff
@@ -472,9 +473,25 @@ public boolean existsById(TableDtoPrimaryKey tableDtoPrimaryKey) {
   @Timed(metricKey = MetricsConstant.REPO_TABLE_DELETE_TIME)
   @Override
   public void deleteById(TableDtoPrimaryKey tableDtoPrimaryKey) {
+    Table table =
+        catalog.loadTable(
+            TableIdentifier.of(
+                tableDtoPrimaryKey.getDatabaseId(), tableDtoPrimaryKey.getTableId()));
+    String location = table.location();
+    String tableCreator = table.properties().get("openhouse.tableCreator");
     catalog.dropTable(
         TableIdentifier.of(tableDtoPrimaryKey.getDatabaseId(), tableDtoPrimaryKey.getTableId()),
-        true);
+        false);
```

Comment:
Why do we still need this boolean? I think functionality wise this change will already impact any overridden implementations of the OpenHouseInternalCatalog until changes are made in the overridden implementations.  And also the boolean is used to determine whether this should be a soft or hard delete. Considering this logic is moved here should we still need the boolean flag?